### ISSUE TITLE
[feature] Add Dockerfile

### ISSUE THREAD

**Issue by NGenetzky** [alpernebbi/git-annex-adapter#7]
Add Dockerfile for easy verification of dependencies and consistent environment for test execution.

**Comment by alpernebbi** [comment]
I don't think I should need or use Docker for this. Dependencies should be verified in `__init__.py`, raising an `ImportError` with a helpful message if not satisfied. Assuming that, tests would either already be running in an environment satisfying minimum requirements or failing as a false negative, urging us to set up dependencies correctly. Ideally, I would use some CI system for running tests.

Personally, I've set up a systemd-nspawn container running Debian, read-only mounted my working repo into it and I run the tests within it. The only reason I do so is that I'm afraid of writing some bad test which mangles the source repo, losing uncommitted work. But this isn't a concern for anyone who isn't working on this library, worst case is you reclone the repo.
 
Despite all that, using Docker could be useful for deploying the metadata gui. I'll have to look into that, but later on.

**Comment by NGenetzky** [comment]
> systemd-nspawn container running Debian, read-only mounted my working repo into it

Sounds interesting.

Well if you change your mind I have a dockerfile I created in my fork that uses the neurodebian package for git annex and is based on `python:3`

**Comment by NGenetzky** [comment]
> Ideally, I would use some CI system for running tests.

Yup, and if you used a docker container you could quickly run it in travis-ci or gitlab-ci with only a few lines of code.
